src/memory.rs

```diff
@@ -84,22 +84,46 @@ pub fn decrement(cell: &mut u8, amount: usize) {
     }
 }
 
-// I/O for brainfuck
+// Cross-platform I/O for brainfuck
+#[cfg(target_os = "windows")]
 extern "C" {
     fn _getch() -> core::ffi::c_char;
     fn _putch(c_char: core::ffi::c_char) -> core::ffi::c_void;
 }
 
 // Read 1 byte input
+#[cfg(target_os = "windows")]
 #[inline]
 fn getch() -> core::ffi::c_char {
-    // SAFETY: trust C
+    // SAFETY: call Windows-specific `_getch`
     unsafe { _getch() }
 }
 
 // Write 1 byte output
+#[cfg(target_os = "windows")]
 #[inline]
 fn putch(c_char: core::ffi::c_char) -> core::ffi::c_void {
-    // SAFETY: trust C
+    // SAFETY: call Windows-specific `_putch`
     unsafe { _putch(c_char) }
 }
+
+// Alternative implementation for non-Windows platforms
+#[cfg(not(target_os = "windows"))]
+#[inline]
+fn getch() -> core::ffi::c_char {
+    use std::io::Read;
+    // Read a single byte from stdin
+    io::stdin()
+        .bytes()
+        .next()
+        .and_then(|result| result.ok())
+        .unwrap_or(0) as core::ffi::c_char
+}
+
+// Alternative implementation for non-Windows platforms
+#[cfg(not(target_os = "windows"))]
+#[inline]
+fn putch(c_char: core::ffi::c_char) {
+    print!("{}", c_char as u8 as char);
+    io::stdout().flush().unwrap();
```

Comment:
Flushing is already handled in `memory::output()`. This is to reduce the amount of times `flush()` is called when printing multiple times at once.